hashicorp/boundary-ui
# Host set permission issue

## Description

- Target does not fetch a 'host' anymore. This is happening now within session.
- Within session, when fetching a 'host', if the user does not have permission we do NOT show an error and let the user quick connect.
- Decouple getting target host sources into host-sets, now isn't part of `model()` but has it's own method `getTargetHostSets`.
- When fetching `host-set`, if the user does not have permissions, we do NOT show it and allow the user for quick connect.
- Decouple getting host from host-sets, now isn't part of `model()` btu has it's own method `getHostsFromHostSets`.
- When fetching `host`, if the user does not have permissions, we do NOT show it and allow the user for quick connect.


## How to Test

- Create a target with address and port.
- Create a host catalog with one host-set and one host.
- Create a host catalog with two host-set and 3 host (2 host in 1 host set, 1 remaining host in other host-set).
- Create a user with specific grants.
- Play around changing `host` and `host-set` grants, allowing JUST list and the app works as expected. Allows quick connect and shows no errors related to grants).
- Play around changing `host` and `host-set` grants, allowing list & read and the app works as expected. Allows quick connect when no host-sets are present, performs quick connect when just one host-set and one host present, shows choose host modal when more than 1 host is present on the target.

Grants used:

- `id=*;type=session;actions=list,read:self,cancel:self`.
- `id=*;type=target;actions=list,authorize-session,read`.
- `id=*;type=host-set;actions=list`.
- `id=*;type=host;actions=list`.
- `id=*;type=host-catalog;actions=list,read`
